### ISSUE TITLE
[Bug]: Watchmode error in makeReport

### ISSUE THREAD

**Issue by seppinho** [epi2me-labs/wf-metagenomics#33]
Hi all,
Awesome pipeline! I'm running the latest version on my Nanopore fastq data using the following command using Nextflow version 22.10.5:
`nextflow run wf-metagenomics/main.nf --fastq test  --database_set "PlusPF-8" --watch_path --batch_size 1`

Everything works for the standard mode but as soon as I add a new fastq file to the folder, the pipeline crashes within the report step ("  ValueError: not enough values to unpack (expected 2, got 0)")

Am I missing something? Thanks for your help!

```
Error executing process > 'kraken_pipeline:makeReport (1)'

Caused by:
  Process `kraken_pipeline:makeReport (1)` terminated with an error exit status (1)

Command executed:

  report.py         "wf-metagenomics-report.html"         --versions versions         --params params.json         --summaries all_stats.1         --lineages "bracken.1"         --vistempl template.html

Command exit status:
  1

Command output:
  (empty)

Command error:
  Traceback (most recent call last):
    File "2023-nanopore-metagenomics/wf-metagenomics/bin/report.py", line 187, in <module>
      main()
    File "2023-nanopore-metagenomics/wf-metagenomics/bin/report.py", line 174, in main
      kraken(args.summaries[0], section)
    File "2023-nanopore-metagenomics/wf-metagenomics/bin/report.py", line 47, in kraken
      lbins, lcounts = list(zip(*len_hist))
  ValueError: not enough values to unpack (expected 2, got 0)

Work dir:
 2023-nanopore-metagenomics/work/18/3e70a6c69ae8aba691b68373851170

Tip: you can replicate the issue by changing to the process work dir and entering the command `bash .command.run`
```

### Operating System

ubuntu 18.04

### Workflow Execution

Command line

### Workflow Execution - CLI Execution Profile

None

### Workflow Version

v2.0.8

### Relevant log output

```shell
[skipped  ] process > kraken_pipeline:unpackTaxonomy                 [100%] 1 of 1, stored: 1 ✔
[skipped  ] process > kraken_pipeline:unpackDatabase                 [100%] 1 of 1, stored: 1 ✔
[e9/e1bb00] process > kraken_pipeline:determine_bracken_length       [100%] 1 of 1 ✔
[89/e934d0] process > kraken_pipeline:kraken_server                  [  0%] 0 of 1
[bf/54d610] process > kraken_pipeline:rebatchFastq (4)               [100%] 4 of 4
[bb/d90787] process > kraken_pipeline:kraken2_client (7)             [  0%] 7 of 4000
[9b/e93d0e] process > kraken_pipeline:progressive_stats (7)          [ 85%] 6 of 7
[aa/9341be] process > kraken_pipeline:progressive_kraken_reports (7) [ 85%] 6 of 7
executor >  local (37)
[skipped  ] process > kraken_pipeline:unpackTaxonomy                 [100%] 1 of 1, stored: 1 ✔
[skipped  ] process > kraken_pipeline:unpackDatabase                 [100%] 1 of 1, stored: 1 ✔
[e9/e1bb00] process > kraken_pipeline:determine_bracken_length       [100%] 1 of 1 ✔
[89/e934d0] process > kraken_pipeline:kraken_server                  [  0%] 0 of 1
[bf/54d610] process > kraken_pipeline:rebatchFastq (4)               [100%] 4 of 4
[bb/d90787] process > kraken_pipeline:kraken2_client (7)             [  0%] 7 of 4000
[9b/e93d0e] process > kraken_pipeline:progressive_stats (7)          [ 85%] 6 of 7
[aa/9341be] process > kraken_pipeline:progressive_kraken_reports (7) [ 85%] 6 of 7
[a4/61917d] process > kraken_pipeline:progressive_bracken (3)        [ 66%] 2 of 3
executor >  local (37)
[skipped  ] process > kraken_pipeline:unpackTaxonomy                 [100%] 1 of 1, stored: 1 ✔
[skipped  ] process > kraken_pipeline:unpackDatabase                 [100%] 1 of 1, stored: 1 ✔
[e9/e1bb00] process > kraken_pipeline:determine_bracken_length       [100%] 1 of 1 ✔
[89/e934d0] process > kraken_pipeline:kraken_server                  [  0%] 0 of 1
[bf/54d610] process > kraken_pipeline:rebatchFastq (4)               [100%] 4 of 4
[bb/d90787] process > kraken_pipeline:kraken2_client (7)             [  0%] 7 of 4000
[9b/e93d0e] process > kraken_pipeline:progressive_stats (7)          [ 85%] 6 of 7
[aa/9341be] process > kraken_pipeline:progressive_kraken_reports (7) [ 85%] 6 of 7
[a4/61917d] process > kraken_pipeline:progressive_bracken (3)        [ 66%] 2 of 3
[a8/e98d8a] process > kraken_pipeline:getVersions                    [100%] 1 of 1 ✔
[a7/fe03de] process > kraken_pipeline:getParams                      [100%] 1 of 1 ✔
[18/3e70a6] process > kraken_pipeline:makeReport (1)                 [ 50%] 1 of 2, failed: 1
[5a/e349bd] process > kraken_pipeline:output (2)                     [100%] 2 of 2
[-        ] process > kraken_pipeline:stop_kraken_server             -
Error executing process > 'kraken_pipeline:makeReport (1)'

Caused by:
  Process `kraken_pipeline:makeReport (1)` terminated with an error exit status (1)

Command executed:

  report.py         "wf-metagenomics-report.html"         --versions versions         --params params.json         --summaries all_stats.1         --lineages "bracken.1"         --vistempl template.html

Command exit status:
  1

Command output:
  (empty)

Command error:
  Traceback (most recent call last):
    File "2023-nanopore-metagenomics/wf-metagenomics/bin/report.py", line 187, in <module>
      main()
    File "2023-nanopore-metagenomics/wf-metagenomics/bin/report.py", line 174, in main
      kraken(args.summaries[0], section)
    File "2023-nanopore-metagenomics/wf-metagenomics/bin/report.py", line 47, in kraken
      lbins, lcounts = list(zip(*len_hist))
  ValueError: not enough values to unpack (expected 2, got 0)

Work dir:
 work/18/3e70a6c69ae8aba691b68373851170

Tip: you can replicate the issue by changing to the process work dir and entering the command `bash .command.run`
```

**Comment by sarahjeeeze** [comment]
Hi, Thanks for reporting. How large each of your Fastq files?

**Comment by Dafnaa** [comment]
hii,

We got the exact same problem. Our Fastq files differ from 1kb to 300 kb. 

Thanks!

**Comment by Dafnaa** [comment]
Hiii,

We got the exact same problems. Our Fastq files are compressed and have very different sizes a few from 1kb to 300kb and some from 7MB.

Did someone found a resolution yet?

Many thanks!

**Comment by sarahjeeeze** [comment]
Hi, Sorry about that, I am still trying to recreate the error. Are you also running on Ubuntu OS or something else?

**Comment by Dafnaa** [comment]
Hi Sarah,

Thanks for your reply. I managed to make it work with our dataset. I basecalled the Fast5 files again using guppy_basecaller. Then I used the guppy_barcoder to put them in the proper barcode folder and then the file sizes were all about 7MB and the work flow worked!  When I use the barcode_kits parameter during guppy_basecaller it also makes the weird smaller fastq files.

And yes I'm using Ubuntu OS.

**Comment by claczny** [comment]
Hi,

I just ran into this problem.

The same workflow worked for two MinION runs (pool1, pool2), but failed on two others (pool3, pool4). These seem to have quite variable sequencing throughput (some samples yielding more than others), i.e., the concatenated pass files for  some barcodes are empty (pool3, pool4). I removed these from the consideration and am running into the same error. Besides the _empty_ files, there are some files that have a very low amount of reads, i.e., low filesizes of 4K, 8K, 12K, 16K, etc. (pool3, pool4).
The smallest from the two successfully processed runs has a filesize of 13M (pool2).

What are your thoughts on this please? :)

Best wishes and stay safe,

Cedric

**Comment by nggvs** [comment]
Hi,

Apologies for the late answer and thank you for using the workflow. The concatenated files that are empty are produced within the wf-metagenomics or is that the input data for the workflow is empty? The original issue posted was already solved in the release [v2.1.0](https://github.com/epi2me-labs/wf-metagenomics/releases/tag/v2.1.0)

Please, let us know if the problem still persist.
